src/backend/base/langflow/components/brightdata/brightdata_structured_data.py

```
@@ -0,0 +1,1108 @@
+        }
+        
+        if all_scores and all_scores[0]["total_score"] >= 15:
+            best_match = all_scores[0]
+            logger.info(f"Auto-detected dataset: {best_match['dataset']} (confidence: {best_match['total_score']})")
+            return best_match["dataset"], best_match["total_score"], detection_details
+        
+        logger.warning(f"Could not auto-detect dataset for URL: {url}")
+        return None, 0, detection_details
+
+    def _calculate_specificity_bonus(self, dataset_name: str, url: str, domain: str, path: str, query: str) -> int:
+        """Calculate specificity bonus based on precise pattern matching"""
```

Ruff (0.11.9)

First line should end with a period, question mark, or exclamation point

Add closing punctuation

(D415)